### PR TITLE
Add option to ignore field for specific table

### PR DESCRIPTION
**Pull request by michealmouner**
add the ability to ignore table field 
options will be like that in config

`    table_ignore_columns:`
`        - student.preferences`
`        - student.creditsExpireAt`

### CODE REVIEW

File: src/SimpleThings/EntityAudit/DependencyInjection/SimpleThingsEntityAuditExtension.php [hunk 1]
```diff
@@ -47,6 +47,7 @@ public function load(array $configs, ContainerBuilder $container)
             'revision_id_field_type',
             'global_ignore_columns',
             'convert_enum_to_string',
+            'ignore_column_for_table',
```
Comment (artursvonda) [inline src/SimpleThings/EntityAudit/DependencyInjection/SimpleThingsEntityAuditExtension.php]: We have `global_ignore_columns` above so maybe follow convention and use `table_ignore_columns`?

File: src/SimpleThings/EntityAudit/EventListener/LogRevisionsListener.php [hunk 2]
```diff
@@ -168,7 +168,6 @@ public function postFlush(PostFlushEventArgs $eventArgs)
                             sprintf('Could not resolve database type for column "%s" during extra updates', $column)
                         );
                     }
-
```
Comment (TamasSzigeti) [inline src/SimpleThings/EntityAudit/EventListener/LogRevisionsListener.php]: I would keep this empty line

File: src/SimpleThings/EntityAudit/EventListener/LogRevisionsListener.php [hunk 3]
```diff
@@ -455,6 +459,11 @@ private function saveRevisionEntityData($class, $entityData, $revType)
                 continue;
             }
 
+            //ignore specific fields for table
+            if(in_array($class->getTableName() . "." . $field , $this->config->getIgnoreColumnForTable())){
```
Comment (TamasSzigeti) [inline src/SimpleThings/EntityAudit/EventListener/LogRevisionsListener.php]: Why is this condition needed 3 times?

Comment (michealmouner) [reply src/SimpleThings/EntityAudit/EventListener/LogRevisionsListener.php]: it exists two times (create migration - insert into audit tables)

Comment (michealmouner) [reply src/SimpleThings/EntityAudit/EventListener/LogRevisionsListener.php]: more details:
this condition exist two times in "insert into audit tables" 
1) ignore this column from SQL statement preparation
2) ignore this column from being calculated as a record change

File: src/SimpleThings/EntityAudit/Resources/config/auditable.xml [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
 <container xmlns="http://symfony.com/schema/dic/services"
     xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-    xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
```
Comment (TamasSzigeti) [inline src/SimpleThings/EntityAudit/Resources/config/auditable.xml]: Why this change?
```suggestion
    xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
```

File: src/SimpleThings/EntityAudit/EventListener/LogRevisionsListener.php [hunk 5]
```diff
@@ -168,8 +168,8 @@ public function postFlush(PostFlushEventArgs $eventArgs)
                             sprintf('Could not resolve database type for column "%s" during extra updates', $column)
                         );
                     }
-                    
                     $types[] = $type;
+
```
Comment (TamasSzigeti) [inline src/SimpleThings/EntityAudit/EventListener/LogRevisionsListener.php]: Not here

File: src/SimpleThings/EntityAudit/EventListener/CreateSchemaListener.php [hunk 6]
```diff
@@ -92,6 +92,11 @@ public function postGenerateSchemaTable(GenerateSchemaTableEventArgs $eventArgs)
         foreach ($entityTable->getColumns() as $column) {
             $columnTypeName = $column->getType()->getName();
             $columnArrayOptions = $column->toArray();
+            // ignore column
+            if(in_array($entityTable->getName()."." . $column->getName() , $this->config->getTableIgnoreColumns())){
```
Comment (TamasSzigeti) [inline src/SimpleThings/EntityAudit/EventListener/CreateSchemaListener.php]: Is adding this logic in 3 places the only way to implement it? Feels a bit scattered. Also, global_ignore_columns work quite differently.

Comment (michealmouner) [reply src/SimpleThings/EntityAudit/EventListener/CreateSchemaListener.php]: yes because of global_ignore_columns ignore updating the column but it creates the column in the audit table
the new table_ignore_columns doesn't create the field on audit table and ignore that field in insert

I think we need it in these three places but we can move the logic in config Class

File: src/SimpleThings/EntityAudit/EventListener/CreateSchemaListener.php [hunk 7]
```diff
@@ -92,8 +92,9 @@ public function postGenerateSchemaTable(GenerateSchemaTableEventArgs $eventArgs)
         foreach ($entityTable->getColumns() as $column) {
             $columnTypeName = $column->getType()->getName();
             $columnArrayOptions = $column->toArray();
-            // ignore column
-            if(in_array($entityTable->getName()."." . $column->getName() , $this->config->getTableIgnoreColumns())){
+
+            //ignore specific fields for table
+            if ($this->config->isIgnoredField($entityTable->getName()."." . $column->getName())) {
```
Comment (TamasSzigeti) [inline src/SimpleThings/EntityAudit/EventListener/CreateSchemaListener.php]: If you set up php quality tools in your IDE, then you notice spacing issues without the pre-commit hooks in place
